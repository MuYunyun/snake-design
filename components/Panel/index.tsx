import * as React from 'react'
import * as cx from 'classnames'

// 通用Panel组件
const getCom = (name?: string) => (props: any) => {
  const { children, className, alignItems, ...ps } = props
  const classNames = cx(className, {
    'snake-design-panel': !name,
    [`snake-design-panel-${name}`]: !!name,
  })
  return <div className={classNames} {...ps}>
    {children}
  </div>
}

export default class Panel extends React.Component<any, any> {

  static Header: any = getCom('header')
  static Footer: any = getCom('footer')
  static Body: any = getCom('body')
  static Addons: any = getCom('addons') // 附带内容

  // A item
  static Item = function(props: any) {
    const {
      leftIcon,
      leftAddon,
      rightIcon,
      rightAddon,
      feedback,
      alignItems,
      children,
      className,
      innerStyle,
      innerClass,
      innerDirection,
      ...ps
    } = props
    const classNames = cx({
      'snake-design-panel-feedback': feedback,
    }, 'snake-design-panel-item', `snake-design-panel-align__${alignItems || 'center'}`, className)
    return <div className={classNames} {...ps}>
      {leftIcon && <div className="snake-design-panel-item-leftIcon">{leftIcon}</div>}
      {leftAddon && <div className="snake-design-panel-item-leftAddon">{leftAddon}</div>}
      <div
        className={cx('snake-design-panel-item-inner', innerClass, {
          [`snake-design-panel-item-inner-${innerDirection}`]: !!innerDirection,
        })}
        style={innerStyle}
      >
        {children}
      </div>
      {rightAddon && <div className="snake-design-panel-item-rightAddon">{rightAddon}</div>}
      {rightIcon && <div className="snake-design-panel-item-rightIcon">{rightIcon}</div>}
    </div>
  }

  render() {
    return getCom()(this.props)
  }
}
