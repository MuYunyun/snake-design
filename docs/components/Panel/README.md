# Panel

基础组件

:::demo

## 简单示例
Panel组件

```require
./components/Panel/simplePanel.tsx
```
:::

## props

### PanelItem参数
| 参数     | 说明              | 类型   | 可选值 | 默认值 |
| -------- | ----------------- | ------ | ------ | ------ |
| leftIcon | 左侧图标 | React.ReactNode | ------ | ------ |
| leftAddon | 左侧插件 | React.ReactNode | ------ | ------ |
| rightIcon | 右侧图标 | React.ReactNode | ------ | ------ |
| rightAddon | 右侧插件 | React.ReactNode | ------ | ------ |
| innerDirection | inner flex 布局方向 | enum | row/column | row |
| feedback | 触摸反馈 | boolean | ------ | ------ |
| innerStyle | 自定义内容样式 | React.CSSProperties | ------ | ------ |
| innerClass | 自定义内容类 | string | ------ | ------ |
| alignItems | label与控件对齐的方式 | enum | stretch/center/end | center |
