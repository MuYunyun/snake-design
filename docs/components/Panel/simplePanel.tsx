import * as React from 'react'

export default class SimpleExample extends React.Component {

  state: any = {
    selectApple: false,
    selectBanana: false,
    bindEleme: false,
    bindMeituan: false,
    bindBaidu: false,
  }

  toggleState = (state: string) => {
    this.setState({
      [state]: !this.state[state],
    })
  }

  render() {
    return (
      <div style={{ background: '#f0f1f2' }}>
        <div>1234</div>
      </div>
    )
  }
}
