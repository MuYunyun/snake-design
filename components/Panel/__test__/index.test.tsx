import * as React from 'react'
import { shallow } from 'enzyme'
import renderToJSON from 'enzyme-to-json'
import Panel from '../index'

describe('components', () => {
  describe('panel', () => {
    const PanelHeader = Panel.Header
    const PanelBody = Panel.Body
    const PanelItem = Panel.Item
    const PanelFooter = Panel.Footer
    const PanelAddons = Panel.Addons

    it('Basic usage of Panel && Panel\'s children com', () => {
      const wrapper = shallow(<Panel>
        <PanelAddons>Addons</PanelAddons>
        <PanelHeader>标题</PanelHeader>
        <PanelBody>
          <PanelItem>Item1</PanelItem>
          <PanelItem>Item2</PanelItem>
        </PanelBody>
        <PanelFooter>底部</PanelFooter>
      </Panel>)

      expect(renderToJSON(wrapper)).toMatchSnapshot()
      expect(wrapper.childAt(1).contains(<PanelHeader>标题</PanelHeader>)).toBe(true)
      expect(wrapper.childAt(2).children().length).toBe(2)
    })
  })
})
