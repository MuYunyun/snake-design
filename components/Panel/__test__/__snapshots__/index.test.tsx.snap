// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components panel Basic usage of Panel && Panel's children com 1`] = `
<div
  className="snake-design-panel"
>
  <Component>
    Addons
  </Component>
  <Component>
    标题
  </Component>
  <Component>
    <Component>
      Item1
    </Component>
    <Component>
      Item2
    </Component>
  </Component>
  <Component>
    底部
  </Component>
</div>
`;
